pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -57,975 +56,801 @@
+        self._amp_type_flag = AMPType.from_str(amp_type)
 
-    @property
-    def has_tpu(self) -> bool:
-        # Here, we are not checking for TPU availability, but instead if User has passed
-        # `tpu_cores` to Trainer for training.
-        if self.tpu_cores is not None:
-            return True
-        return self._map_devices_to_accelerator(_AcceleratorType.TPU)
-
-    @property
-    def use_tpu(self) -> bool:
-        return self._accelerator_type == _AcceleratorType.TPU and self.has_tpu
-
-    @property
-    def tpu_id(self) -> Optional[int]:
-        if self.use_tpu and isinstance(self.tpu_cores, list):
-            return self.tpu_cores[0]
-        return None
+        if amp_level is not None and self._amp_type_flag != AMPType.APEX:
+            raise MisconfigurationException(
+                f"You have asked for `amp_level={amp_level!r}` but it's only supported with `amp_backend='apex'`."
+            )
+        self._amp_level_flag = amp_level
 
-    @property
-    def has_ipu(self) -> bool:
-        # Here, we are not checking for IPU availability, but instead if User has passed
-        # `ipus` to Trainer for training.
-        if self.ipus is not None or isinstance(self._strategy, IPUStrategy):
-            return True
-        return self._map_devices_to_accelerator(_AcceleratorType.IPU)
+    def _check_device_config_and_set_final_flags(
+        self,
+        devices: Optional[Union[List[int], str, int]],
+        num_nodes: int,
+        num_processes: Optional[int],
+        gpus: Optional[Union[List[int], str, int]],
+        ipus: Optional[int],
+        tpu_cores: Optional[Union[List[int], int]],
+    ) -> None:
+        if num_nodes == "auto":
+            self._num_nodes_flag = 1
+        else:
+            self._num_nodes_flag = int(num_nodes) if num_nodes is not None else 1
+
+        self._device_flag = devices
+
+        # TODO: Delete this parsing section when num_processes, gpus, ipus and tpu_cores get removed
+        self._gpus = gpus
+        self._tpu_cores = tpu_cores
+        gpus = device_parser.parse_gpu_ids(gpus)
+        tpu_cores = device_parser.parse_tpu_cores(tpu_cores)
+        deprecated_devices_specific_flag = num_processes or gpus or ipus or tpu_cores
+        if deprecated_devices_specific_flag and deprecated_devices_specific_flag not in (0, "0"):
+            self._map_deprecated_devices_specfic_info_to_accelerator_and_device_flag(
+                devices, deprecated_devices_specific_flag, num_processes, gpus, ipus, tpu_cores
+            )
 
-    @property
-    def use_ipu(self) -> bool:
-        return self._accelerator_type == _AcceleratorType.IPU and self.has_ipu
+        if self._device_flag == "auto" and self._accelerator_flag is None:
+            raise MisconfigurationException(
+                f"You passed `devices={devices}` but haven't specified"
+                " `accelerator=('auto'|'tpu'|'gpu'|'ipu'|'cpu')` for the devices mapping"
```

Comment:
```suggestion
                " `accelerator=('auto'|'tpu'|'gpu'|'ipu'|'cpu')` for the devices mapping."
```

Reply:
@four4fish @ananthsub  Should accelerator be mapped to `auto` if not provided or not raise an expectation here as it is watched lines 161 ?